Discover EKS: handle invalid Auth Mode and Private only endpoint

marcoandredinis opened the pull request:
During the EKS Enrollment in the Discover wizard, when listing clusters, check for pre-reqs when showing available clusters.

1. When in Cloud, only allow enrollment if the cluster has a Public Endpoint for configuration.

1. Only show clusters which have the API authentication mode enabled (either API only or API_AND_CONFIG_MAP).

kimlisa commented on web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EksClustersList.tsx:
```diff
@@ -121,26 +123,51 @@ function getStatus(item: CheckedEksCluster) {
+    };
+  }
+
+  if (['failed', 'deleting'].includes(item.status)) {
```
i think failed state should go on 161? if sometime failed, try again?

kimlisa commented on web/packages/teleport/src/Discover/Kubernetes/EnrollEKSCluster/EksClustersList.tsx:
```diff
@@ -121,26 +123,51 @@ function getStatus(item: CheckedEksCluster) {
+    };
+  }
+
+  if (item.authenticationMode === 'CONFIG_MAP') {
+    return {
+      disabled: true,
+      disabledText:
+        'Authentication mode is CONFIG_MAP, only API and API_AND_CONFIG_MAP are supported.',
+    };
+  }
```
should we do `!= API && != API_AND_CONFIG_MAP` isntead?

marcoandredinis replied:
Given that we only set 3 possible values, I don't think it would be any different. But I'll change it to be more explicit about the valid values 👍

marcoandredinis committed:
Discover EKS: handle invalid Auth Mode and Private only endpoint

During the EKS Enrollment in the Discover wizard, when listing clusters,
check for pre-reqs when showing available clusters.

1. When in Cloud, only allow enrollment if the cluster has a Public
   Endpoint for configuration.

1. Only show clusters which have the API authentication mode enabled
   (either API only or API_AND_CONFIG_MAP).
